### DISCUSSION TITLE
Testing artifact release

### DISCUSSION THREAD

**Discussion by stefanofornari** [shred/acme4j#162] [General]
Hello Richard,
thanks so much for this great project. I am developing a java command line to wrap the main functionality of acme4j and I would love to use the testing provider you use in your project. Would it be possible for your to release it as an additional artifact to be added as test dependency to a third project?

Thanks in advance,
Stefano

PS: let me know if you are interested in having a command line interface integrated in acme4j

**Comment by shred** [comment]
Hi Stefano!

I'm not quite sure what you mean.

For the integration tests, I use [Pebble](https://github.com/letsencrypt/pebble) and the pebble-challtestsrv. Both are 3rd party tools that are provided by Let's Encrypt as Docker image.

To connect to Pebble, I use the pebble provider which is part of the core `acme4j-client` module.

To connect and control the challtestsrv, I use a tiny client that is part of `acme4j-it`. You can use this client by adding `acme4j-it` as test dependency to your project.

It's okay for me if your command line interface is a separate project. Once it's completed, I will gladly link to it in the documentation.

**Comment by stefanofornari** [comment]
Hi Richard,
thanks for your feedback. What I mean is fundamentally TestableConnectionProvider and all classes needed to use it. If I had a acme4j-test jar to add to my dependency I could use them like:

```
<dependency>
    <groupId>org.shredzone.acme4j</groupId>
    <artifactId>acme4j-test</artifactId>
    <version>3.4.0</version>
    <scope>test</scope>
</dependency>
```

It's for unit testing, not integration testing. Today I have no other choices than including the source code.

**Comment by shred** [comment]
I see what you mean now. Yes, this topic is already discussed in issue #64, as you found out. I will close this discussion in favor of the issue.
